beamer/chain.py

```diff
@@ -77,6 +77,10 @@ def stop(self) -> None:
         self._stop = True
         self._thread.join(_STOP_TIMEOUT)
 
+    def subscribe(self, event_processor: "EventProcessor") -> None:
+        self._on_new_events.append(event_processor.add_events)
+        self._on_sync_done.append(event_processor.mark_sync_done)
```

Comment:
This is not ideal because it introduces coupling between `EventMonitor` and `EventProcessor`, but let's go with it for now. We can think of a nice way to get rid of the coupling after the PR is merged.